streaming/replication.c

```diff
@@ -0,0 +1,1931 @@
+                , rd->divisor
+                , rrddim_flag_check(rd, RRDDIM_FLAG_OBSOLETE)?"obsolete":""
+                , rrddim_flag_check(rd, RRDDIM_FLAG_HIDDEN)?"hidden":""
+                , rrddim_flag_check(rd, RRDDIM_FLAG_DONT_DETECT_RESETS_OR_OVERFLOWS)?"noreset":""
+        );
+        rd->exposed = 1;
+    }
+
+    // send the chart local custom variables
+    RRDSETVAR *rs;
+    for(rs = st->variables; rs ;rs = rs->next) {
+        if(unlikely(rs->type == RRDVAR_TYPE_CALCULATED && rs->options & RRDVAR_OPTION_CUSTOM_CHART_VAR)) {
+            calculated_number *value = (calculated_number *) rs->value;
+
+            buffer_sprintf(
+                    rep_state->build
+                    , "VARIABLE CHART %s = " CALCULATED_NUMBER_FORMAT "\n"
+                    , rs->variable
+                    , *value
+            );
+        }
+    }
+
+    st->upstream_resync_time = st->last_collected_time.tv_sec + (remote_clock_resync_iterations * st->update_every);
+}
+// send all metrics from all the charts for this gap
+void sender_fill_gap_nolock(REPLICATION_STATE *rep_state, RRDSET *st, GAP a_gap)
+{
+    RRDDIM *rd;
+    struct rrddim_query_handle handle;
+    // TODO: Add it in the stream.conf file
+    unsigned int block_size_in_bytes = MEM_PAGE_BLOCK_SIZE;
+    unsigned int sample_in_bytes = (unsigned int) sizeof(storage_number);
+    unsigned int default_replication_gap_block_size = block_size_in_bytes/sample_in_bytes; // in samples
+
+    unsigned int block_id = 0;
+    int update_every = st->update_every;
+
+    time_t t_delta_start = a_gap.t_window.t_start;
+    time_t t_delta_first = a_gap.t_window.t_first;
+    time_t t_delta_end = a_gap.t_window.t_end;
+    time_t window_start, window_end;
+    time_t newest_connection = st->rrdhost->sender->t_newest_connection;
+    UNUSED(t_delta_first);
+   
+    // Chop the GAP time interval to fit a MEM_PAGE_BLOCK_SIZE(4096)
+    // window_end is more important than window start
+    // window_end = MAX((t_delta_end + (t_delta_end % update_every)), newest_connection);
+    window_end = MAX((t_delta_end + (update_every - (t_delta_end % update_every))), newest_connection);
+    window_start = t_delta_start - (t_delta_start % update_every);
+    size_t replication_points = (t_delta_end - t_delta_start) / update_every + 1;
+    if (replication_points > default_replication_gap_block_size){
+        replication_points = default_replication_gap_block_size;
+        window_start = window_end - replication_points * update_every;
+    }
+
+    debug(
+        D_REPLICATION,
+        "%s: REP GAP timestamps request from MEMORY\nt_d_start: %ld, t_d_end:  %ld \nw_start:  %ld w_end:  %ld \nreplication points: %lu\nnewest_connection: %ld\nupdate_every: %d",
+        REPLICATION_MSG,
+        t_delta_start,
+        t_delta_end,
+        window_start,
+        window_end,
+        replication_points,
+        newest_connection,
+        update_every);
+
+    char gap_uuid_str[UUID_STR_LEN];
```

Comment:
Here we have again a variable called `gap_uuid_str`, I observed that only one variable has the size `UUID_STR_LEN + 1`, while the others have length `UUID_STR_LEN`, please, remove the `+1` already described, because UUID_STR_LEN is already defined with the additional character as you can see [here](https://github.com/netdata/netdata/blob/8b0fffe33bccf950496f5ffee358320bde71e5b5/libnetdata/libnetdata.h#L96-L98).